random-beans/src/test/java/io/github/benas/randombeans/PopulatorImplTest.java

```diff
@@ -145,6 +146,23 @@ public void javaNetTypesShouldBePopulated() throws Exception {
         assertThat(website.getUrl()).isNotNull();
     }
 
+    @Test(expected = BeanPopulationException.class)
+    public void failsToPopulateInterfacesAndAbstractClassesIfScanClasspathForConcreteClassesIsDisabled() throws Exception {
+        Populator nonScanningPopulator = aNewPopulatorBuilder().scanClasspathForConcreteClasses(false).build();
+
+        nonScanningPopulator.populateBean(Mamals.class);
+    }
+
+    @Test
+    public void generatesConcreteTypesForInterfacesAndAbstractClassesIfScanClasspathForConcreteClassesIsEnabled() throws Exception {
+        Populator scanningPopulator = aNewPopulatorBuilder().scanClasspathForConcreteClasses(true).build();
+
+        Mamals mamals = scanningPopulator.populateBean(Mamals.class);
+
+        assertThat(mamals.getMamal()).isOfAnyClassIn(Human.class, Ape.class, Person.class, SocialPerson.class);
+        assertThat(mamals.getMamalImpl()).isOfAnyClassIn(Human.class, Ape.class, Person.class, SocialPerson.class);
+    }
+
```

Comment:
Nice catch :), at the moment this fails with:

>Caused by: io.github.benas.randombeans.api.BeanPopulationException: Unable to generate a random instance of type class org.apache.commons.lang3.builder.DiffBuilder
>Caused by: java.lang.ClassCastException: sun.reflect.generics.reflectiveObjects.ParameterizedTypeImpl cannot be cast to java.lang.Class

I will try add support for this.